### PR TITLE
docs: API updates to reflect reality

### PR DESCRIPTION
**Pull request by robert-ancell**

### CODE REVIEW

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by robert-ancell** [comment]
I've attempted to document all the fields as returned from snapd, some of them I'm not sure what they mean exactly so please check if the descriptions could be improved.

**Comment by mvo5** [comment]
add to whitelist

**Comment by mvo5** [comment]
Thanks for this update! :+1:

**Comment by zyga** [comment]
Looks good

File: docs/rest.md [hunk 1]
```diff
@@ -225,14 +229,20 @@ snaps is supported via the `refresh` key.
 
 [//]: # keep the fields sorted, both in the description and the sample above. Makes scanning easier
 
-* `description`: snap description
+* `channel`: which channel the package is currently tracking.
```
Comment (niemeyer) [inline docs/rest.md]: s/package/snap/

File: docs/rest.md [hunk 2]
```diff
@@ -225,14 +229,20 @@ snaps is supported via the `refresh` key.
...
+* `confinement`: the confinement used; one of `strict` or `devmode`.
```
Comment (niemeyer) [inline docs/rest.md]:
```
* `confinement`: the confinement requested by the snap itself; one of `strict` or `devmode`.
* `devmode`: whether the snap is currently installed in development mode
```

File: docs/rest.md [hunk 3]
```diff
@@ -225,14 +229,20 @@ snaps is supported via the `refresh` key.
...
+* `description`: snap description.
+* `developer`: developer who created the snap.
 * `download-size`: how big the download will be.
 * `icon`: a url to the snap icon, possibly relative to this server.
+* `id`: Unique ID for this snap
```
Comment (niemeyer) [inline docs/rest.md]: s/U/u/

Comment (niemeyer) [inline docs/rest.md]: and s/$/./, looks like the common end of line here.

File: docs/rest.md [hunk 4]
```diff
@@ -225,14 +229,20 @@ snaps is supported via the `refresh` key.
...
 * `name`: the snap name.
 * `prices`: JSON object with properties named by ISO 4217 currency code. The values of the properties are numerics representing the cost in each currency. For free snaps, the "prices" property is omitted.
+* `private`: true if this is not a publically available snap.
```
Comment (niemeyer) [inline docs/rest.md]:
```
* `private`: whether this snap is available to its author only or public
```

File: docs/rest.md (hunk 2 above)
Comment (niemeyer) [inline docs/rest.md]: Sorry, devmode should go below. This is the results from find only.

File: docs/rest.md [hunk 5]
```diff
@@ -303,12 +319,12 @@ In addition to the fields described in `/v2/find`:
 
 [//]: # keep the fields sorted!
 
-* `channel`: which channel the package is currently tracking.
+* `apps`: JSON array of apps the snap provides. Each app has a `name` field to name a binary this app provides.
 * `installed-size`: how much space the snap itself (not its data) uses.
 * `install-date`: the date and time when the snap was installed.
-* `status`: can be either `installed` or `active` (i.e. is current).
+* `trymode`: true if the app was installed in try mode.
```
Comment (niemeyer) [inline docs/rest.md]: Please copy the devmode description here.

Also, why was status dropped? Note that the description differs.

**Comment by niemeyer** [comment]
THanks!

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Pull request merged**